localstack/services/cloudwatch/alarm_scheduler.py

```diff
@@ -127,7 +127,8 @@ def get_metric_alarm_details_for_alarm_arn(alarm_arn: str) -> Optional[MetricAla
 
 def get_cloudwatch_client_for_region_of_alarm(alarm_arn: str) -> "CloudWatchClient":
     region = arns.extract_region_from_arn(alarm_arn)
-    return connect_to(region_name=region).cloudwatch
+    account_id = arns.extract_account_id_from_arn(alarm_arn)
+    return connect_to(region_name=region, aws_access_key_id=account_id).cloudwatch
```

Comment:
Both `extract_region_from_arn` and `extract_account_id_from_arn` call `parse_arn`

Consider using `parse_arn` once which will be more efficient.